### PR TITLE
Redirect blog.wellcomelibrary.org to Internet Archive

### PR DESCRIPTION
**Pull request by kenoir**
https://blog.stage.wellcomelibrary.org

TODO:
- [x] Some more unit / e2e tests required 
- [x] Import blog.wellcomelibrary.org DNS

In a future PR:
- Update internal DNS for blog.stage.wellcomelibrary.org (requested REQ0021077)
- Confirm final crawl and set the correct URL prefix in the lambda.
- Deploy updated lambda to prod & point blog.wellcomelibrary.org -> at CloudFront
- Update internal DNS for blog.wellcomelibrary.org

### CODE REVIEW

File: cloudfront/wellcomelibrary.org/edge-lambda/src/wellcomeLibraryBlogRedirect.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    _: Context
+) => {
+    const request: CloudFrontRequest = event.Records[0].cf.request;
+
+    const blogRedirect = redirectToBlog(request);
+    if (blogRedirect) {
+        return blogRedirect;
+    }
+
+    // If we've matched nothing so far then set the host header for Wellcome Library
```
Comment (kenoir) [inline cloudfront/wellcomelibrary.org/edge-lambda/src/wellcomeLibraryBlogRedirect.ts:32]: We should probably just error if we reach here.

File: cloudfront/wellcomelibrary.org/edge-lambda/src/wellcomeLibraryRedirect.ts [hunk 2]
```diff
@@ -76,6 +76,11 @@ export const requestHandler = async (
 ) => {
   const request: CloudFrontRequest = event.Records[0].cf.request;
 
+  if (request.headers.host?.length === 1) {
```
Comment (kenoir) [inline cloudfront/wellcomelibrary.org/edge-lambda/src/wellcomeLibraryRedirect.ts:79]: Remove.

File: cloudfront/wellcomelibrary.org/edge-lambda/testRedirects.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+async function testRedirects(tests: RedirectTest) {
+  Object.entries(tests.pathTests).map(async ([fromPath, to]) => {
+    const from = `${tests.host}${fromPath}`;
+
+    try {
+      const axiosResponse = await axios.get(from);
+      const responseUrl = axiosResponse.request.res.responseUrl;
+      if (responseUrl === to) {
+        console.info(chalk.green(`✓ ${from} === ${responseUrl}`));
```
Comment (jamieparkinson) [inline cloudfront/wellcomelibrary.org/edge-lambda/testRedirects.ts:26]: How come these tests aren't using jest? Feels a bit like reinventing the wheel to print these assertion outcomes...

File: cloudfront/wellcomelibrary.org/terraform/blog.tf [hunk 4]
```diff
@@ -0,0 +1,49 @@
+module "wellcomelibrary_blog-prod" {
+  source = "./cloudfront_distro"
+
+  distro_alternative_names = [
+    "blog.wellcomelibrary.org"
+  ]
+  acm_certificate_arn = module.cert-stage.arn
+
+  origins = [{
```
[the remark this answers is not in the document]
Comment (jamieparkinson) [reply cloudfront/wellcomelibrary.org/terraform/blog.tf:9]: `blog-archive.wellcomelibrary.org`?

File: cloudfront/wellcomelibrary.org/edge-lambda/testRedirects.ts (hunk 3 above)
Comment (kenoir) [reply cloudfront/wellcomelibrary.org/edge-lambda/testRedirects.ts:26]: It did cross my mind. I'm not sure it makes sense to use jest here as these aren't really testing the lambda, (but they really are?). I'll have a think.

Comment (jamieparkinson) [reply cloudfront/wellcomelibrary.org/edge-lambda/testRedirects.ts:26]: It doesn't matter all that much, but I would ask why whether they're testing the lambda should change whether you use a test runner (I'd say it shouldn't).

Comment (kenoir) [reply cloudfront/wellcomelibrary.org/edge-lambda/testRedirects.ts]: I really don't have a strong feeling about this - other than maybe next time i will _try_ using a test runner for slightly non-conventional tests.

**Comment by kenoir** [comment]
I am merging this as I want to merge something else and this version of the PR is not in public view (and will be iterated on before we do the public release).

**Pull request merged**